### PR TITLE
Add while loop section to parallel model

### PR DESCRIPTION
**Pull request by jdahm**
Describes the syntax and limitation of while loops.

### CODE REVIEW

File: GTScript-Parallel-model.md [hunk 1]
```diff
@@ -289,3 +289,56 @@ with computation(...):
...
+            if mask[i, j]:
+                c[i, j, k_] += 1
+                a[i, j, k_] += 1
+            mask[i, j] = (a[i, j, k_] < b[i, j, k_])
+```
+
+The conclusion from this is that the user is not allowed to write to
+variables used with a horizontal offset either in the mask or elsewhere in
+the body of the while. The gtscript frontend implemented in gt4py contains checks
+to ensure that a user cannot write code incompatible with this restriction.
```
Comment (havogt) [inline GTScript-Parallel-model.md:344]:
```suggestion
```

File: GTScript-Parallel-model.md [hunk 2]
```diff
@@ -289,3 +289,56 @@ with computation(...):
...
+indices could execute the statements a different number of iterations. The syntax is:
+
+```python
+with computation(FORWARD), interval(...):
+    while a < b:
+        c += 1
+        a += 1
+```
+
+In the parallel model the user will expect this to translate to
```
Comment (havogt) [inline GTScript-Parallel-model.md:307]:
```suggestion
This translates to
```

**Review by havogt — CHANGES_REQUESTED** [review]
Thanks! This summarizes what I was missing to review the PR.

I was thinking a bit more about it and came to the conclusion, that the restriction actually doesn't apply, because it is already covered by rule 4. Therefore it would be good to actually add proper checking for restrictions described here (which probably are currently missing) and remove the restriction on the while loop in the implementation.
Please double check if I got it right or if I missed an important that makes it special.

**Comment by havogt** [comment]
The important point is, if I am not mistaken, that the restriction doesn't apply to temporaries and therefore doesn't apply to the mask. Does it make sense?

File: GTScript-Parallel-model.md (hunk 1 above)
Comment (jdahm) [reply GTScript-Parallel-model.md:344]: > within a computation, it is illegal to write to an external (non-temporary) field (or aliases pointing to the same memory location) if it is also read with horizontal offset; this rule does not apply to temporaries,

True, but I believe there still could be issues with the update of the mask. Consider this:

```python
a = input_field # a is a temporary field
while a[1, 0, 0] < 0:
    a += 1
    ...
```
I suppose the rule should be that there are no horizontal offsets (or vertical offsets for that matter) in the while condition. Does that sound right to you?

**Review by havogt — APPROVED** [review]
Looks good.

**Pull request merged**